### DISCUSSION TITLE
PaperScroller scroll() & zoom

### DISCUSSION THREAD

**Discussion by frnora** [clientIO/joint#2272] [Q&A] [answered]
### Introduction

Hi,

I am trying to scroll the paperscroller to a point but when zoom is not 1, the point is off.

`scroll - Try to scroll the paper scroller so that the position (x,y) on the paper (in local coordinates) is at the center of the paper scroller viewport.`
Is it really in '**local coordinates**'? What does that even mean, I thought local coordinates were independent on zoom factor (I thought that would be paper coordinates in that case).

`isPointVisible` correctly returns whether the point is visible or not, even if zoomed in (as far as I could test). But I cannot scroll to the same point with the same zoom. Looking at the `scrollToElement` I can see you are accounting for the scale/zoom. Does this need to be done for a point, too? And can't some paper.XpointToYpoint be used? If yes, which one? ( `localToPaperPoint` should do it, right? - but in that case is `scroll` really accepting a 'local' point and not a 'paper' point?)

Thank you.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment, marked as answer]
Sorry for the inconvenience. This is a known issue that will be addressed in version `4.0` (breakthrough change).

Like any other method `scroll()` should receive a point in the local coordinate system. Currently, it receives a point that should be scaled but not translated (the case where the point `0,0` is not in the top left corner of the paper).

```js
const localPoint = new g.Point(1000, 0); // untransformed coordinates, the same as graph coordinates
const scale = paperScroller.zoom(); // current zoom level
const scrollPoint = localPoint.scale(scale, scale);
paperScroller.scroll(scrollPoint.x, scrollPoint.y);
```
